stripped/colabs/Detekcja_sezonowosci_stripped.py

```diff
@@ -0,0 +1,915 @@
+  - attribute (str): nazwa kolumny będąca zmienną zależną (dependent variable),
+  - year_split (int): rok rozdzielający zbiór treningowy i testowy,
+  - p, d, q (int): parametry funkcji SARIMAX, argumentu order,
+  - P, D, Q, s (int): parametry funkcji SARIMAX, argumantu seasonal_order.
+  """
+  assert attribute in df.columns, f"The attribute '{attribute}' is not a column in the DataFrame."
+  # Stworzenie zbioru treningowego i testowego
+  def train_test_split(df: pd.DataFrame, year_split: int) -> None:
+    """
+    Funckja dzieli ramke danych na zbiór treningowy i testowy.
+    - df (Pandas DataFrame): ramka danych w Pandas zawierająca co najmniej następujace kolumny: Year, Month, atrybut,
+    - year_split (int): rok rozdzielający zbiór treningowy i testowy.
+    """
+    train_data =  df.loc[df['Year'] < year_split]
+    test_data = df.loc[df['Year'] >= year_split]
+    return train_data, test_data
+  train_data, test_data = train_test_split(df, year_split)
+  # Sprawdzenie stacjonarności przy użyciu funkcji perform_augumented_dickey_fuller_print_result
+  if perform_augumented_dickey_fuller_print_result(df, attribute, False) == "Stationary":
+    enforce_stat = False
+  else:
+    enforce_stat = True
+  # Trenowanie modelu
+  def train_model(train_df: pd.DataFrame, attribute: str, p:int, d:int, q:int, P:int, D:int, Q:int, s:int, enforce_stat: bool):
+    """
+    Funkcja, która trenuje model SARIMAX.
+    Parametry:
+    - train_df (Pandas DataFrame): zbiór treningowy,
+    - attribute (str): nazwa kolumny będąca zmienną zależną (dependent variable),
+    - year_split (int): rok rozdzielający zbiór treningowy i testowy,
+    - p, d, q (int): parametry funkcji SARIMAX, argumentu order,
+    - P, D, Q, s (int): parametry funkcji SARIMAX, argumantu seasonal_order,
+    - enforce_stat (bool): parametr determinujący argument enforce_stationarity funkcji SARIMAX.
+    """
+    sarima = SARIMAX(train_df[attribute],
+                order=(p,d,q),
+                seasonal_order=(P,D,Q,s),
+                trend="ct",
+                enforce_stationarity=enforce_stat).fit()
+    return sarima
+  sarima = train_model(train_data, attribute, p, d, q, P, D, Q, s, enforce_stat)
+  # Test modelu i wyświetlenie wyników
+  def model_predict(model, year_split: int):
+    """
+    Funkcja dokonuje predykcji za pomocą wczytanego modelu począwszy od pierwszego stycznia roku year_split do 2023-09-01.
+    - model: model za pomocą, którego dokonujemy predykcji,
+    - year_split (int): rok rozdzielający zbiór treningowy i testowy.
+    """
+    df_pred = model.predict(start= str(year_split) + "-01-01", end="2023-09-01", steps=len(test_data))
+    return df_pred
+  df_pred = model_predict(sarima, year_split)
+  # Wyświetlenie metryk
+  def print_mean_squerd_error_and_r2_score(test_df: pd.DataFrame, test_pred: pd.DataFrame, attribute: str) -> None:
```

Comment:
Niepotrzebne osadzenie we wnetrzu funkcji

Reply:
Poprawione